test/mocks/MockBaseActionsRouter.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.24;
+
+import {Currency} from "@uniswap/v4-core/src/types/Currency.sol";
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
+import {BaseActionsRouter} from "../../src/base/BaseActionsRouter.sol";
+import {Actions} from "../../src/libraries/Actions.sol";
+import {ReentrancyLock} from "../../src/base/ReentrancyLock.sol";
+import {Locker} from "../../src/libraries/Locker.sol";
+
+contract MockBaseActionsRouter is BaseActionsRouter, ReentrancyLock {
+    uint256 public swapCount;
+    uint256 public increaseLiqCount;
+    uint256 public decreaseLiqCount;
+    uint256 public donateCount;
+    uint256 public clearCount;
+    uint256 public settleCount;
+    uint256 public takeCount;
+    uint256 public mintCount;
+    uint256 public burnCount;
+
+    constructor(IPoolManager poolManager) BaseActionsRouter(poolManager) {}
+
+    function executeAction(bytes calldata params) external isNotLocked {
+        _executeActions(params);
+    }
+
+    function _handleAction(uint256 action, bytes calldata params) internal override {
```

Comment:
ooo nice ok!!! so now the implementer just chooses their actions, and they can keep this and override/revert with the default actions or roll their own.

Guess we should discuss for Router/Posm how we wanna do this? Ie if router is going to have an addLiq/removeLiq command then we can make them the same value